src/wallet/wallet.cpp

```diff
@@ -2349,14 +2349,19 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
     size_t max_descendants = (size_t)std::max<int64_t>(1, limit_descendant_count);
     bool fRejectLongChains = gArgs.GetBoolArg("-walletrejectlongchains", DEFAULT_WALLET_REJECT_LONG_CHAINS);
 
-    bool res = value_to_select <= 0 ||
+    bool res = value_to_select <= 0;
+    if (res) {
+        bnb_used = false;
+    } else {
+        res = value_to_select <= 0 ||
```

Comment:
Why leave this case here when we know it is false?

Reply:
Oops, forgot to delete that.